### ISSUE TITLE
cabal publish errors

### ISSUE THREAD

**Issue by alunduil** [freebroccolo/docker-haskell#84]
I'm using this docker image in Google's cloud build system to do automated publishing to hackage but I get the following error message when running `cabal publish --upload`:

```
Already have image (with digest): haskell
cabal: The builtin HTTP implementation does not support HTTPS, but using HTTPS
for authenticated uploads is recommended. The transport implementations with
HTTPS support are curl, wget, powershellbut they require the corresponding
external program to be available. You can either make one available or use
plain HTTP by using the global flag --http-transport=plain-http (or putting
the equivalent in the config file). With plain HTTP, your password is sent
using HTTP digest authentication so it cannot be easily intercepted, but it is
not as secure as using HTTPS.
```

I see that curl is removed from the image.  Is there any reason we can't leave it in the image so that publish works from this environment?

**Comment by psftw** [comment]
@alunduil thank you for reporting. I've updated the Dockerfiles (https://github.com/freebroccolo/docker-haskell/commit/03467e1a14543d83d33833e669249a3c42f7b7c8) but there will still be a delay before the new images get published (and will include a GHC 8.6.3 version bump).  Let me know if you run in to problems as I don't actually have a way to verify `cabal publish` myself.

**Comment by psftw** [comment]
https://github.com/docker-library/official-images/pull/5263
